test/examples/ucp_hello_world.c

```diff
@@ -188,6 +188,7 @@ static int run_ucx_client(ucp_worker_h ucp_worker)
     } else if (UCS_PTR_STATUS(request) != UCS_OK) {
         printf("UCX address message was scheduled for send\n");
         wait(ucp_worker, request);
+        request->completed = 0; /* Reset request state before recycling it */
```

Comment:
Is it workaround?
I expected the ucp_params.request_init callback function should be called by UCX before get request from the pool and use it.